Update buttons and text

zliving opened the pull request:
This branch turned into more of a button text and location update than the connection for the url builder. Chris and I are going to rearchitect the way the buttons work.

RayGar committed:
Updated the background and added a new atlas with modified pressed buttons. They looked bad around the corners.

RayGar committed:
Added an mp3 file to the assets folder. Created and called the Music instance to play the mp3 file in the Play screen.

zliving committed:
Add GameOverLobby Screen

zliving committed:
Setup GameOverLobby

zliving committed:
Add GameOverLobby to main

zliving committed:
Add correctly answered questions

zliving committed:
Count the number of correct answers

zliving committed:
Add gameover loby transition

RayGar committed:
New atlas and background color

Added a new atlas that has a set of buttons with a colored border and a light gray center, pressed and not pressed, and changed the color of the background to 64 - 64 - 64

zliving committed:
Update Button Location

zliving committed:
Merged branch feature/add-button-logic into feature/connect-url-builder

zliving committed:
Smooth out text

Text isn't so pixelated.

zliving committed:
Add comment for linear filter

RayGar reviewed (commented):
Suggestion: you may remove "final atlas 4.png" and "final atlas 4.pack" in favor of the "final atlas 5" files as everything in the final atlas 4 is contained in it plus the latest batch of buttons.

RayGar committed:
Removed unused assets.

cvasqu09 commented on android/assets/final atlas 4.pack:
```diff
@@ -0,0 +1,210 @@
+
+final atlas 4.png
```
Can you get rid of this pack and the png since we already have the 5th atlas.

cvasqu09 commented on core/src/main/java/Scenes/BaseScreen.java:
```diff
@@ -57,7 +59,7 @@
     // Creates an atlas object which can use all the textures within it. Each screen will have
     // access to the atlas in order to create button skins from it.
-    buttonAtlas = new TextureAtlas(Gdx.files.internal("buttonAtlas.pack"));
+    buttonAtlas = new TextureAtlas(Gdx.files.internal("final atlas 4.pack"));
```
Just update this with the "final atlas 5" once the 4th one is deleted.

cvasqu09 commented on core/src/main/java/UIElements/TriviaButtonBuilder.java:
```diff
@@ -46,7 +46,7 @@
   protected static final float WORLD_WIDTH = 480;
   protected static final float WORLD_HEIGHT = 800;
-  private static final float xButtonPosition = 50;
+  private static final float xButtonPosition = (480-300)/2;
```
Just change this to 90?

jmtaber129 replied:
Or change it to "(WORLD_WIDTH - 300) / 2" if that's what the 480 represents.

cvasqu09 reviewed (changes requested):
Clean up some of the unused assets and change the line to use the new atlas.

jmtaber129 commented on core/src/main/java/InputListeners/TriviaButtonListener.java:
```diff
@@ -49,6 +49,7 @@ public boolean touchDown(InputEvent event, float x, float y, int pointer, int bu
     // Check if the answer is correct and there hasn't been a click registered yet.
     if(selectedAnswer.compareTo(correctAnswer) == 0 && !questionSetButtons.get(buttonNumber).isDisabled()) {
       scoreManager.setPlayerScore(10*(int)(15-triviaButtonBuilder.timePassed));
+      scoreManager.correctlyAnswered = scoreManager.correctlyAnswered++;
```
Shouldn't this be a single increment?  I haven't checked it, but I would think that the post increment operation would evaluate to the current value, leaving the variable unchanged.  E.g., if correctlyAnswered = 5, then correctlyAnswered++ evaluates to 5, then gets incremented to 6, but then it gets set to 5 (since that's what the right-hand operator evaluates to).  I could be wrong, but please validate this either way.

jmtaber129 commented on core/src/main/java/Scenes/BaseScreen.java:
```diff
@@ -35,6 +36,7 @@
   protected Skin buttonSkin;
   private FreeTypeFontGenerator generator;
   private FreeTypeFontParameter parameter;
+  //public Music music;
```
Remove this and the unused import.

jmtaber129 commented on core/src/main/java/Scenes/BaseScreen.java:
```diff
@@ -138,7 +141,10 @@ protected BitmapFont generateNewFont(String fontPath, int size, Color color) {
-    return generator.generateFont(parameter);
+    BitmapFont smoothFont = generator.generateFont(parameter);
+    // Linearly filter the text to prevent pixelation. 
+    smoothFont.getRegion().getTexture().setFilter(Texture.TextureFilter.Linear, Texture.TextureFilter.Linear);
```
Line length > 100.  Run auto-formatter.

jmtaber129 commented on core/src/main/java/Scenes/GameOverLobby.java:
```diff
@@ -0,0 +1,37 @@
+package Scenes;
+
+import com.badlogic.gdx.graphics.Color;
```
Remove unused import.

jmtaber129 commented on core/src/main/java/Scenes/GameOverLobby.java:
```diff
@@ -0,0 +1,37 @@
+    // Creates GlyphLayout to get width for centering text in the banner.
+    bannerTextGlyphLayout = new GlyphLayout(bannerText, "Game Over");
+    // Calculate the center for the text to be drawn in the banner.
+    glyphCenterX = ((int) WORLD_WIDTH - (int) bannerTextGlyphLayout.width) / 2;
```
Cast to int after the division operation.

jmtaber129 commented on core/src/main/java/Scenes/Play.java:
```diff
@@ -5,8 +5,11 @@
 import com.badlogic.gdx.graphics.g2d.GlyphLayout;
 import com.badlogic.gdx.scenes.scene2d.InputEvent;
 import com.badlogic.gdx.scenes.scene2d.InputListener;
+import com.badlogic.gdx.audio.Sound;
```
Remove unused import.

jmtaber129 commented on core/src/main/java/Scenes/Play.java:
```diff
@@ -33,6 +37,20 @@ public Play(FilePlayMain mainGame) {
+    /* Credit for the composition goes to Ludwig van Beethoven.
+       Credit for the performance goes to Simone Rezi.
+       Credit for hosting the file goes to musopen.org.
+     */
```
Does this need to be right here, and not somewhere else?  If it absolutely has to be here, then either fix multiline comment formatting, or change to single-line (implementation) comment formatting (preferred).

jmtaber129 commented on core/src/main/java/Scenes/Play.java:
```diff
@@ -83,4 +101,11 @@ public boolean touchDown(InputEvent event, float x, float y, int pointer, int bu
-}
+
+
+  // Disposes the music so that it only plays during gameplay.
```
Move this comment inside the method, i.e. above music.dispose().

jmtaber129 commented on core/src/main/java/UIElements/TriviaButtonBuilder.java:
```diff
@@ -106,6 +106,11 @@ public TriviaButtonBuilder() {
     yButtonPosition -= questionButton.getHeight()/3;
     correctAnswerButton = new TextButton(currentQuestion.correctAnswer, textButtonStyle);
+    correctAnswerButton.getLabelCell().width(300);
+    correctAnswerButton.setWidth(300);
```
It seems like 300 represents the button width, and is used as such several times.  Consider making it a constant.

jmtaber129 commented on core/src/main/java/UIElements/TriviaButtonBuilder.java:
```diff
@@ -117,6 +122,11 @@ public TriviaButtonBuilder() {
+      incorrectAnswerButton.setWidth(300);
+      incorrectAnswerButton.getLabelCell().center();
+      incorrectAnswerButton.getLabel().setWrap(true);
+      incorrectAnswerButton.invalidate();
```
There seems to be some code duplication between correctAnswerButton and the incorrectAnswerButtons.  Consider pulling the previous 5-6 lines into a private helper method.

jmtaber129 reviewed (changes requested):
generateButtons() in TriviaButtonBuilder is getting pretty long (>70 lines).  In my experience/opinion, you generally want to keep methods relatively short (<40 lines).  Any longer and you should consider pulling code into private helper methods.  This helps with readability.

Might have additional comments once merge conflicts are resolved.

RayGar committed:
Added one atlas for the roulette stage animation, and another one for the update. Also added the code needed for the a simple looping rotating animation in the play screen.

cvasqu09 committed:
Added the roulette screen

RayGar committed:
Updated the atlas in a way that writes the rings in the correct order, and fixed the outlines. Also modified the speed of the animation.

RayGar committed:
Removed the cause of the "jerkiness" in the png's from the rings atlas.

RayGar committed:
Added a new atlas with a skull and a triangle to be used for the roulette stage animation.

cvasqu09 committed:
Copied code for the ring animation.

cvasqu09 committed:
Added the new atlas and roulette asset.

cvasqu09 committed:
Added the safe button and the skull button to the roulette stage.

cvasqu09 committed:
Added when to stop when a file is lost.

cvasqu09 committed:
Added the stopping frame for the safe button.

cvasqu09 committed:
Added roulette button to main menu and tweaked the timing for the wheel.

cvasqu09 committed:
Made frame times final

cvasqu09 committed:
Formatting changes and added comments

zliving committed:
Merge branch 'feature/roulette-stage'

Conflicts:
	core/src/main/java/Scenes/BaseScreen.java
	core/src/main/java/com/mygdx/game/FilePlayMain.java

zliving committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/background-update'

Conflicts:
	core/src/main/java/Scenes/BaseScreen.java
	core/src/main/java/Scenes/Play.java

zliving committed:
Merge remote-tracking branch 'refs/remotes/origin/develop'

Conflicts:
	core/src/main/java/Scenes/MainMenu.java

zliving committed:
Updates
